### DISCUSSION TITLE
What's the difference between pg and p for listing things in current dir?

### DISCUSSION THREAD

**Discussion by aspieln3r** [xonsh/xonsh#4463] [Q&A] [answered]
exact difference is not given in the tutorial

**Comment by gforsyth** [comment, marked as answer]
Hey @aspieln3r -- could you clarify the question?  I'm not exactly sure what you're asking.

**Reply by aspieln3r** [reply]
I'm asking about this part in https://xon.sh/tutorial.html
>Similarly to regex globbing, normal globbing can be performed (either in Python mode or subprocess mode) by using the g``

which is also used later

>\>\>p\`.*\`
[Path('foo'), Path('bar')]
>\>\>[x for x in pg\`**\` if x.is_symlink()]
[Path('a_link')]

can you tell me what difference do g\` \` make when resolving path or difference between normal globbing and regexp globbing?

**Reply by gforsyth** [reply]
Ahhh, gotcha!

using only backticks `` is a regular expression match, so any files that match the regex inside are returned.

g`` is a glob match, so follows the rules of Python's `glob.glob`

```
(base)   ~/tmp/glob 🐚  ls
AbA a   aa  ab  ba
(base)   ~/tmp/glob 🐚  `a*`  # regex for 0 or more `a`
['a', 'aa']
(base)   ~/tmp/glob 🐚  g`a*`  # glob for anything beginning with `a`
['AbA', 'a', 'aa', 'ab']
```

Putting a `p` in front of _either_ of them returns the same results, but with those results as `pathlib.Path` objects, so:

```
(base)   ~/tmp/glob 🐚  p`a*`
[PosixPath('a'), PosixPath('aa')]
(base)   ~/tmp/glob 🐚  pg`a*`
[PosixPath('AbA'), PosixPath('a'), PosixPath('aa'), PosixPath('ab')]
```
